Formula/djbdns.rb

```diff
@@ -28,6 +28,10 @@ class Djbdns < Formula
   def install
     inreplace "hier.c", 'c("/"', "c(auto_home"
     inreplace "dnscache-conf.c", "/etc/dnsroots", "#{etc}/dnsroots"
+    # Work around build error from root requirement: "Oops. Your getgroups() returned 0,
+    # and setgroups() failed; this means that I can't reliably do my shsgr test. Please
+    # either ``make'' as root or ``make'' while you're in one or more supplementary groups."
+    inreplace "Makefile", "( cat warn-shsgr; exit 1 )", "cat warn-shsgr" if OS.linux?
```

Comment:
Can't we use fakeroot for this?

Reply:
I didn't realize we've just fixed `fakeroot`, but I just tried system `fakeroot`, `make`, `setup`, `check` and it worked!  So that does seem like a simpler way to fix this.  I think we should only use `fakeroot` on Linux for now as it's not needed on macOS.  Perhaps as a follow up to this we should also see if the other packages we patched like this will also work with `fakeroot` now, though that should be in a separate PR.

Reply:
This should be ready to go now.